### PR TITLE
Add usePureComponent option

### PR DESCRIPTION
**Pull request by ahoseinian**

### CODE REVIEW

File: src/translate.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import hoistStatics from 'hoist-non-react-statics';
+import shallowEqual from 'fbjs/lib/shallowEqual';
```
[the remark this answers is not in the document]
Comment (ahoseinian) [reply src/translate.js]: As we are already dependent to `React` and react is dependent to `fbjs` I thought this is ok! however if you think copying shallowEqual is a better option I can change it?

[the remark this answers is not in the document]
Comment (jamuhl) [reply src/translate.js]: runs also in preact -> that's why pulling in a dependency of react is a bad idea

[the remark this answers is not in the document]
Comment (ahoseinian) [reply src/translate.js]: oh I see

**Comment by jamuhl** [comment]
would it be possible to add a test case? so we get coverage up again...regarding the codeclimate that shallowEqual file could be excluded in the codeclimate.yml

**Comment by jamuhl** [comment]
published with react-i18next@7.2.0

**Pull request merged**